api/src/devcomp/domain/usecases/verify-and-save-answer.js

```diff
@@ -0,0 +1,36 @@
+import { NotFoundError } from '../../../shared/domain/errors.js';
+import { PassageDoesNotExistsError } from '../errors.js';
+
+async function verifyAndSaveAnswer({
+  userResponse,
+  elementId,
+  passageId,
+  passageRepository,
+  moduleRepository,
+  elementAnswerRepository,
+}) {
+  const passage = await _getPassage({ passageId, passageRepository });
+  const module = await moduleRepository.getBySlugForVerification({ slug: passage.moduleId });
+  const element = module.getElementById(elementId);
+  const grain = module.getGrainByElementId(elementId);
+  const { correction, userResponseValue } = element.assess(userResponse);
+  return await elementAnswerRepository.save({
```

Comment:
Eviter les `return await`, juste faire `return` (sinon ça fait des promise dans des promise)

@aceol tu saurais retrouver le talk que tu nous avais partagé stp ?

Reply:
@Steph0 il existe une règle ESLint qui explique cela très bien (et au passage que nous pourrions activer) : https://eslint.org/docs/latest/rules/no-return-await

Reply:
@lego-technix merci du lien :) C'était ce compromis qui ne me plaisait pas trop dans le fait de ne pas `return await` :( 

> You can avoid the extra microtask by not awaiting the return value, **with the trade off of the function no longer being a part of the stack trace if an error is thrown asynchronously from the Promise being returned. This can make debugging more difficult.**